File: src/net/merayen/elastic/ui/objects/top/views/filebrowserview/FileBrowserView.java
package net.merayen.elastic.ui.objects.top.views.filebrowserview;

import java.io.File;

import net.merayen.elastic.ui.objects.top.views.View;

public class FileBrowserView extends View {
	private FileBrowserViewBar bar = new FileBrowserViewBar();
	private FileList file_list = new FileList();

	@Override
	public void onInit() {
		super.onInit();
		add(file_list);
		file_list.browse("/");
		file_list.getTranslation().x = 0;
		file_list.getTranslation().y = 20;
		file_list.setHandler(new FileList.Handler() {
			
			@Override
			public void onSelect(File file) {
				if(file == null)
					file_list.up();
				else if(file.isDirectory())
					file_list.browse(file.getAbsolutePath());
			}
		});

		add(bar);
	}

	@Override
	public void onUpdate() {
		file_list.width = width - 20;
		file_list.height = height;
	}

	@Override
	public View cloneView() {
		return new FileBrowserView();
	}

}

File: src/main/kotlin/net/merayen/elastic/ui/objects/top/views/filebrowserview/FileBrowserView.kt
package net.merayen.elastic.ui.objects.top.views.filebrowserview

import net.merayen.elastic.ui.objects.top.easymotion.Branch
import net.merayen.elastic.ui.objects.top.views.View
import java.io.File

class FileBrowserView : View() {
	private val bar = FileBrowserViewBar()
	private val fileList = FileList()

	override fun onInit() {
		super.onInit()
		add(fileList)
		fileList.browse("/")
		fileList.translation.x = 0f
		fileList.translation.y = 40f
		fileList.setHandler(object : FileList.Handler {
			override fun onSelect(file: File?) {
				if (file == null)
					fileList.up()
				else if (file.isDirectory)
					fileList.browse(file.absolutePath)
			}

		})

		add(bar)
	}

	override fun onUpdate() {
		fileList.layoutWidth = getWidth()
		fileList.layoutHeight = getHeight() - 40
	}

	override fun cloneView(): View {
		return FileBrowserView()
	}

	override val easyMotionBranch = object : Branch(this) {}
}
